Modelica/Mechanics/MultiBody/Frames.mo

```diff
@@ -3076,9 +3076,10 @@ Optionally, residues are returned as well if <code>withResidues=true</code>.
         "Orientation object to rotate frame 1 into frame 2";
     protected
       Real e_x[3](each final unit="1")=if length(n_x) < 1e-10 then {1,0,0} else normalize(n_x);
-      Real n_z_aux[3](each final unit="1")=cross(e_x, n_y);
-      Real n_y_aux[3](each final unit="1")=if n_z_aux*n_z_aux > 1.0e-6 then n_y
-             else (if abs(e_x[1]) > 1.0e-6 then {0,1,0} else {1,0,0});
+      Real e_y[3](each final unit="1")=if length(n_y) < 1e-10 then {0,1,0} else normalize(n_y);
+      Real n_z_aux[3](each final unit="1")=cross(e_x, e_y);
+      Real n_y_aux[3](each final unit="1")=if n_z_aux*n_z_aux > 1.0e-6 then e_y
```

Comment:
@MartinOtter As the length of e_y is always 1, `if n_z_aux*n_z_aux > 1.0e-6` only assures the two input vectors aren't collinear now. Could we simply change it to `if length(n_z_aux) > 1e-3`?

Reply:
n_z_aux*n_z_aux avoids the sqrt operation of function length. Though sqrt converges fast, I prefer the current notation.

Reply:
I understand. We could apply the same rule on `if length(n_x) < 1e-10 then ...` as well since the 'length' is not really needed there. But having a threshold of 1e-20 would probably lead to numerical problems.